### ISSUE TITLE
Certain shaders in slangp collection failing

### ISSUE THREAD

**Issue by star69rem** [SnowflakePowered/librashader#147]
My program includes all the the shaders from this collection:
https://github.com/libretro/slang-shaders

Originally, people just used crt-lottes and crt-royale, but I recently added a file browser that let people choose any slangp profile from that collection. Some of the slang profiles just output a black screen with Librashader. If I can collect a list of the ones that don't work, would you be interested in verifying those?

**Comment by chyyran** [comment]
Yes, please let me know which shaders don’t work.

**Comment by star69rem** [comment]
The following return a non null libra_error_t when calling d3d11_filter_chain_create_deferred:

0x0000021add68bf10 {...}

1) vector-glow
2) mamehlsh
3) crt-maximus-royale
4) crt-maximus-royale-half-res-mode
5) crt-maximus-royale-fast-mode
6) crt-guest-advanced-ntsc

These don't throw an exception but only output a black screen. This might be okay though since they're labeled "slangtest," maybe that's just how they behave.

1) crt-slangtest-lanczos
2) crt-slangtest-cubic

**Comment by chyyran** [comment]
![crt-maximus-royale](https://github.com/user-attachments/assets/bf38ebc2-d16d-4b91-8919-532b5aadd38c)
![crt-maximus-royale-half-res-mode](https://github.com/user-attachments/assets/31a5df6c-29de-48e2-bc18-fc8bcae36877)
![crt-maximus-royale-fast-mode](https://github.com/user-attachments/assets/46db4817-069c-4581-8581-14b9472eeee9)
![crt-guest-advanced-ntsc](https://github.com/user-attachments/assets/20d52efd-1b97-4e4f-8f18-b9b9277e2347)
![vector-glow](https://github.com/user-attachments/assets/a0ea9749-1fcb-46c4-a5b9-a2c84650aecf)
![mame_hlsl](https://github.com/user-attachments/assets/017d1c0d-4904-4cd1-a65c-095faeed4afb)


All of those shaders seem to be working fine when testing with librashader-cli. 

```
librashader-cli render -p .\test\shaders_slang\crt\$SHADERP.slangp -i .\triangle.png -o $SHADERP.png -r d3d11
```
The `slangtest` ones work as well, but they don't seem to do anything super obvious.

![crt-slangtest-lanczos](https://github.com/user-attachments/assets/887ac75a-c4e5-444b-94b3-151866a3e051)
![crt-slangtest-cubic](https://github.com/user-attachments/assets/4685df68-9eae-421a-b3de-a1db208939b8)

**Comment by chyyran** [comment]
Did a sanity check with the latest [ares nightly](https://github.com/ares-emulator/ares/actions/runs/11557300101) which uses the C API on Metal, all of the listed shaders work fine. Likely that whatever issues you have are D3D11 specific but without more details it's hard to tell what's going on.

**Comment by star69rem** [comment]
> Did a sanity check with the latest [ares nightly](https://github.com/ares-emulator/ares/actions/runs/11557300101) on macOS which uses the C API on Metal, all of the listed shaders work fine. Likely that whatever issues you have are D3D11 specific but without more details it's hard to tell what's going on. Are you using librashader 0.5.1?

Looks like some of them are failing on preset_create. Seems to be some kind of file path or missing file issue.

L"D:\\finalfightmister\\Final Fight Remake\\assets\\shaders\\crt\\mame_hlsl.slangp"

PresetError(IOError("\\\\?\\D:\\finalfightmister\\Final Fight Remake\\assets\\shaders\\crt\\shaders\\mame_hlsl\\shaders\\lut.slang", Os { code: 2, kind: NotFound, message: "The system cannot find the file specified." })): There was an error parsing the preset.

This is in mame_hlsl.slangp, but it doesn't seem to be included in that archive that I linked from github. So maybe it's just missing files.

shader11 = shaders/mame_hlsl/shaders/lut.slang
filter_linear11 = true

**Comment by chyyran** [comment]
lut.slang was commented out in the current iteration of the slang-shaders repository. Maybe try re-pulling to get an up to date archive?

**Comment by star69rem** [comment]
Yup, that was with it. Sorry, I foolishly assumed that everything in that shader collection worked. Thought it was a little bit more mature than that. Didn't realize they were still making basic fixes like that.

**Comment by chyyran** [comment]
RenderDoc dumps would help a lot for the slangtest shaders.
